ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx

```diff
@@ -145,7 +155,15 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
             <PublishOnDemandForm
               mode="git"
               formData={data}
-              setFormData={(newData) => setState(newData)}
+              setFormData={(newData) => {
+                setState(newData);
+                const mergedData = { ...data, ...newData };
```

Comment:
What's this for? Doesn't newData come with comment, commidIds and pub target?

Reply:
newData is an object with only the edited field, for example `{ comment: 'a comment' }`